examples/textbooks/README.md

```diff
@@ -0,0 +1,106 @@
+# Guidelines
+
+Here you can find the code used to generate large synthetic datasets like [Cosmopedia](https://huggingface.co/datasets/HuggingFaceTB/cosmopedia). You need to have a dataset containing prompts, in this case we're using [cosmopedia-100k](https://huggingface.co/datasets/HuggingFaceTB/cosmopedia-100k).
+
+## Setup
+Since we want to generate a large volume of textbooks and the generations might take a long time, we save the intermediate generations in `checkpoint_path` and track the progress and throughput with `wandb`.
+```bash
+pip install wandb
+wandb init
+```
+
+## Generation
+To run the generations on the first 2000 prompts on 2 TGI instances, you can use:
+```bash
+python ./examples/textbooks/generate_syntehtic_textbooks.py \
+    --model mistralai/Mixtral-8x7B-Instruct-v0.1 \
+    --instances 2 \
+    --prompts_dataset "HuggingFaceTB/cosmopedia-100k" \
+    --prompt_column prompt \
+    --max_samples 2000 \
```

Comment:
Maybe add a comment here saying `--max_samples -1 ` means generating for the whole dataset.